detectable/src/main/java/com/synopsys/integration/detectable/detectables/cargo/parse/CargoLockParser.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * detectable
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detectable.detectables.cargo.parse;
+
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.moandjiezana.toml.Toml;
+import com.synopsys.integration.bdio.graph.DependencyGraph;
+import com.synopsys.integration.bdio.graph.MutableDependencyGraph;
+import com.synopsys.integration.bdio.graph.MutableMapDependencyGraph;
+import com.synopsys.integration.bdio.model.Forge;
+import com.synopsys.integration.bdio.model.dependency.Dependency;
+import com.synopsys.integration.bdio.model.externalid.ExternalId;
+import com.synopsys.integration.bdio.model.externalid.ExternalIdFactory;
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+import com.synopsys.integration.detectable.detectables.cargo.model.CargoLock;
+import com.synopsys.integration.detectable.detectables.cargo.model.Package;
+
+public class CargoLockParser {
+
+    private final ExternalIdFactory externalIdFactory = new ExternalIdFactory();
+
+    private final Map<String, Dependency> packageMap = new HashMap<>();
+
+    public DependencyGraph parseLockFile(final InputStream cargoLockInputStream) throws DetectableException {
+        try {
+            final CargoLock cargoLock = new Toml().read(cargoLockInputStream).to(CargoLock.class);
```

Comment:
Does this Toml library require an `InputStream`? If it can take a `String` we should do that instead. If not then this is fine.

Reply:
Turns out it can, good call